spec/dependency_container_spec.rb

```diff
@@ -0,0 +1,119 @@
+require "spec_helper"
+
+describe Nurse::DependencyContainer do
+  let(:container) do
+    container = Nurse::DependencyContainer.new
+
+    container.define(Hash) do
+      Hash.new
+    end
+
+    container.define(:definition) do
+      :defined_object
+    end
+  end
+
+  describe "#define" do
+    it "defines dependencies" do
+      container.define(:foo) do
+        :bar
+      end
+
+      container.get(:foo).must_equal :bar
+    end
+
+    it "passess the container as argument for the block" do
+      container.define(:hash) do |di|
+        {name: di.get(:name)}
+      end
+
+      container.define :name do
+        "marcelo"
+      end
+
+      container.get(:hash)[:name].must_equal("marcelo")
+    end
+
+    it "throws an exception if dependency was already defined" do
+      begin
+        container.define(Hash)
+        fail
+      rescue Nurse::DependencyContainer::DependencyAlreadyDefined => e
+        e.message.must_equal "Dependency 'Hash' was already defined"
+      end
+    end
+  end
+
+  describe "#define!" do
+    it "overrites definition" do
+      container.define!(:dependency) { :bar }
+      container.get(:dependency).must_equal :bar
+
+      container.define!(:dependency) do |di|
```

Comment:
Unused block argument - `di`. You can omit the argument if you don't care about it.